src/main/java/edu/cornell/kfs/concur/businessobjects/ValidationResult.java

```diff
@@ -59,8 +61,28 @@ public String getErrorMessagesAsOneFormattedString() {
                 result.append(KFSConstants.NEWLINE);
             }
         }
-        
-        return result.toString();
+
+        return addMessageHeaderAndTruncate(result.toString(), ConcurConstants.VALIDATION_RESULT_MESSAGE_MAX_LENGTH);
+    }
+
+    private String addMessageHeaderAndTruncate(String message, int maxLength) {
+        String errorMessagesString = addMessageHeader(message);
+        errorMessagesString = truncateMessageLength(errorMessagesString, maxLength);
```

Comment:
Curious about this. I realize there are message limits that need to be adhered to and why you are doing this, but is there an understanding that there could be more errors than what is being returned due to the truncation being performed?  Meaning, will the users somehow understand that they could go through this validation cycle multiple times just because they were not given all the errors the first time?